chain/network/src/peer_manager/network_state/routing.rs

```diff
@@ -133,7 +150,10 @@ impl NetworkState {
     ) -> Result<PeerId, FindRouteError> {
         match target {
             PeerIdOrHash::PeerId(peer_id) => {
-                self.graph.routing_table.find_next_hop_for_target(peer_id)
+                match self.graph.routing_table.find_next_hop_for_target(peer_id) {
+                    Ok(peer_id) => Ok(peer_id),
+                    Err(_) => self.graph_v2.routing_table.find_next_hop_for_target(peer_id),
+                }
```

Comment:
This behavior is here in anticipation of a release in which we deprecate the old graph. We would reach an intermediate state in which not-yet-upgraded nodes are running both versions while upgraded nodes are running only V2. In such case graph_v2 will have better reachability than graph (due to upgraded nodes not broadcasting their edges in the old style), and this logic would allow the not-yet-upgraded nodes to inter-op smoothly.

Reply:
I think hitting `graph` first and only falling back to `graph_v2` if necessary is the more safe approach. Suppose we release this version and realize something is wrong with `graph_v2`, for example it produces some forwarding decisions sending messages in loops. We can expect everything to continue running smoothly because `graph` is producing the routing decisions. Meanwhile we'll be able to inspect via debug api/ui to see that `graph_v2` is doing something wrong. If we detect such a scenario we can also patch `graph_v2` without a need to revert anything urgently.

And yes, good question, we are not changing any aspect of peer discovery or selection. The network is built the same way as before. The way we figure out how to route through the resulting network is being changed.